ui/tests/integration/components/recovery/page/snapshots-test.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'vault/tests/helpers';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+import { GENERAL } from 'vault/tests/helpers/general-selectors';
+import { dateFormat } from 'core/helpers/date-format';
+
+const SELECTORS = {
+  badge: (name) => `[data-test-badge="${name}"]`,
+};
+
+module('Integration | Component | recovery/snapshots', function (hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.model = {
+      snapshots: [],
+      canLoadSnapshot: false,
+    };
+    this.version = this.owner.lookup('service:version');
+    this.version.type = 'enterprise';
+  });
+
+  test('it displays empty state in CE', async function (assert) {
+    this.version.type = 'community';
+    await render(hbs`<Recovery::Page::Snapshots @model={{this.model}}/>`);
+    assert
+      .dom(GENERAL.emptyStateTitle)
+      .hasText('Secrets Recovery is an enterprise feature', 'CE empty state title renders');
+    assert
+      .dom(GENERAL.emptyStateMessage)
+      .hasText(
+        'Secrets Recovery allows you to restore accidentally deleted or lost secrets from a snapshot. The snapshots can be provided via upload or loaded from external storage.',
+        'CE empty state message renders'
+      );
+    assert
+      .dom(GENERAL.emptyStateActions)
+      .hasText('Learn more about upgrading', 'CE empty state action renders');
+
+    assert.dom(SELECTORS.badge('enterprise')).exists();
+  });
+
+  test('it displays empty state in non root namespace', async function (assert) {
+    this.nsService = this.owner.lookup('service:namespace');
+    this.nsService.path = 'parent1/child1';
```

Comment:
I think this is the only setup necessary for this test? Curious why we need to stub the endpoint below